src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateExecution.java

```diff
@@ -91,4 +126,42 @@ private String computePathOnDisk(String physicalPathOnDisk) {
         FilePath diskFilePath = new FilePath(new File(physicalPathOnDisk));
         return new FilePath(diskFilePath, run.getParent().getFullName() + "/" + run.getNumber()).getRemote();
     }
+
+    /**
+     * Find the upstream job that matches the given upstream name.
+     *
+     * @param upstreamJob the upstream job name
+     * @return the upstream job with the given name if any, {@code null} otherwise
+     */
+    @CheckForNull
+    private Item findUpstreamJob(String upstreamJob) {
```

Comment:
I referenced https://github.com/jenkinsci/promoted-builds-plugin/pull/62, which describes the implementation.
IMHO for a PoC it's better to just use a full name. Maybe I'll create a new API plugin in the short-term